### DISCUSSION TITLE
eval cue from cue tool?

### DISCUSSION THREAD

**Discussion by proppy** [cuelang/cue#397] [General]
Is there a builtin to "eval" a CUE string from cue scripting (tooling) layer?

**Comment by mpvl** [comment]
Not yet. But there should be an `pkg/encoding/cue`.

**Reply by verdverm** [reply]
With the `text/template` and a `cueencoding.Unmarshal`, it seems rather tirvial to implement recursion and programming.

@mpvl is there a point at which we stop caring about that Turing completeness can be created within Cue by roundabout methods? (and rely on just recommending not to do)

**Comment by mikelnrd** [comment]
I didn't realise cue had the new Discussions beta feature turned on.... otherwise I wouldn't have posted about pkg/encoding/cue here: https://github.com/cuelang/cue/issues/423

Just linking it here in case it's helpful

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/397.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
